core/sail/federation/src/main/java/org/eclipse/rdf4j/sail/federation/optimizers/EmptyPatternOptimizer.java

```diff
@@ -58,7 +70,8 @@ public void meet(StatementPattern node)
 		Value obj = node.getObjectVar().getValue();
 		Resource[] ctx = getContexts(node.getContextVar());
 		for (RepositoryConnection member : members) {
-			if (member.hasStatement(subj, pred, obj, true, ctx)) {
+			RepositoryBloomFilter bloomFilter = bloomFilters.apply(member.getRepository());
+			if (bloomFilter == null || bloomFilter.mayHaveStatement(member, subj, pred, obj, ctx)) {
```

Comment:
This implementation detail, that a bloom filter is necessary for all repositories or else optimisation does not occur, should be documented in the class javadoc comment.